File: Treachery-iOS/Treachery-iOS/Managers/AnalyticsService.swift
//
//  AnalyticsService.swift
//  Treachery-iOS
//
//  Created by Luke Solomon on 3/24/26.
//

import FirebaseAnalytics

enum AnalyticsService {

    static func trackScreen(_ screenName: String) {
        Analytics.logEvent(AnalyticsEventScreenView, parameters: [
            AnalyticsParameterScreenName: screenName
        ])
    }

    static func trackEvent(_ name: String, params: [String: Any]? = nil) {
        Analytics.logEvent(name, parameters: params)
    }

    static func setUserId(_ uid: String?) {
        Analytics.setUserID(uid)
    }

    static func setUserProperties(_ properties: [String: String]) {
        for (key, value) in properties {
            Analytics.setUserProperty(value, forName: key)
        }
    }
}

File: TreacheryAndroid/app/src/main/java/com/solomon/treachery/data/AnalyticsService.kt
package com.solomon.treachery.data

import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.analytics.logEvent

object AnalyticsService {
    private var analytics: FirebaseAnalytics? = null

    fun initialize(analytics: FirebaseAnalytics) {
        this.analytics = analytics
    }

    fun trackScreen(screenName: String) {
        analytics?.logEvent(FirebaseAnalytics.Event.SCREEN_VIEW) {
            param(FirebaseAnalytics.Param.SCREEN_NAME, screenName)
        }
    }

    fun trackEvent(eventName: String, params: Map<String, String> = emptyMap()) {
        analytics?.logEvent(eventName) {
            params.forEach { (key, value) ->
                param(key, value)
            }
        }
    }

    fun setUserId(userId: String?) {
        analytics?.setUserId(userId)
    }

    fun setUserProperties(properties: Map<String, String>) {
        properties.forEach { (key, value) ->
            analytics?.setUserProperty(key, value)
        }
    }
}
